21/09/2014 01:51:58.715] - [INFO] - Client starting
21/09/2014 01:51:59.080] - [INFO] - [id: 0x928ac41c] OPEN
21/09/2014 01:51:59.107] - [INFO] - [id: 0x928ac41c, /127.0.0.1:3259 => localhost/127.0.0.1:19999] BOUND: /127.0.0.1:3259
21/09/2014 01:51:59.109] - [INFO] - [id: 0x928ac41c, /127.0.0.1:3259 => localhost/127.0.0.1:19999] CONNECTED: localhost/127.0.0.1:19999
21/09/2014 01:51:59.200] - [INFO] - [id: 0x928ac41c, /127.0.0.1:3259 :> localhost/127.0.0.1:19999] DISCONNECTED
21/09/2014 01:51:59.203] - [INFO] - [id: 0x928ac41c, /127.0.0.1:3259 :> localhost/127.0.0.1:19999] UNBOUND
21/09/2014 01:51:59.209] - [INFO] - [id: 0x928ac41c, /127.0.0.1:3259 :> localhost/127.0.0.1:19999] CLOSED
21/09/2014 01:51:59.282] - [SEVERE] - Unexpected exception from upstream. java.nio.channels.ClosedChannelException

21/09/2014 01:55:47.489] - [INFO] - Client starting
21/09/2014 01:55:47.644] - [INFO] - [id: 0x8217d15e] OPEN
21/09/2014 01:55:47.664] - [INFO] - [id: 0x8217d15e, /127.0.0.1:3268 => localhost/127.0.0.1:19999] BOUND: /127.0.0.1:3268
21/09/2014 01:55:47.669] - [INFO] - [id: 0x8217d15e, /127.0.0.1:3268 => localhost/127.0.0.1:19999] CONNECTED: localhost/127.0.0.1:19999
21/09/2014 01:55:48.593] - [INFO] - writing response back for turn 1 for player 1
21/09/2014 01:55:48.602] - [INFO] - message sent: took 66,198,226
21/09/2014 01:57:32.539] - [SEVERE] - Exception thrown in getMove on turn 2
java.lang.NullPointerException
	at GameMap.bfs(GameMap.java:254)
	at GameMap.whichPath(GameMap.java:112)
	at PlayerAI.getMove(PlayerAI.java:56)
	at ClientHandler.messageReceived(ClientHandler.java:85)
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
	at ClientHandler.handleUpstream(ClientHandler.java:40)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:296)
	at org.jboss.netty.handler.codec.oneone.OneToOneDecoder.handleUpstream(OneToOneDecoder.java:70)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:296)
	at org.jboss.netty.handler.codec.frame.FrameDecoder.unfoldAndFireMessageReceived(FrameDecoder.java:462)
	at org.jboss.netty.handler.codec.frame.FrameDecoder.callDecode(FrameDecoder.java:443)
	at org.jboss.netty.handler.codec.frame.FrameDecoder.messageReceived(FrameDecoder.java:303)
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:559)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:268)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:255)
	at org.jboss.netty.channel.socket.oio.OioWorker.process(OioWorker.java:71)
	at org.jboss.netty.channel.socket.oio.AbstractOioWorker.run(AbstractOioWorker.java:73)
	at org.jboss.netty.channel.socket.oio.OioWorker.run(OioWorker.java:51)
	at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108)
	at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

21/09/2014 01:57:32.547] - [INFO] - writing response back for turn 2 for player 1
21/09/2014 01:57:32.553] - [SEVERE] - Unexpected exception from upstream. java.net.SocketException: Software caused connection abort: socket write error

21/09/2014 01:57:32.557] - [INFO] - [id: 0x8217d15e, /127.0.0.1:3268 :> localhost/127.0.0.1:19999] DISCONNECTED
21/09/2014 01:57:32.560] - [INFO] - [id: 0x8217d15e, /127.0.0.1:3268 :> localhost/127.0.0.1:19999] UNBOUND
21/09/2014 01:57:32.562] - [INFO] - [id: 0x8217d15e, /127.0.0.1:3268 :> localhost/127.0.0.1:19999] CLOSED
21/09/2014 01:57:32.564] - [INFO] - message sent: took 103,836,729,298
21/09/2014 01:58:01.416] - [INFO] - Client starting
21/09/2014 01:58:01.615] - [INFO] - [id: 0x2679910f] OPEN
21/09/2014 01:58:01.648] - [INFO] - [id: 0x2679910f, /127.0.0.1:3279 => localhost/127.0.0.1:19999] BOUND: /127.0.0.1:3279
21/09/2014 01:58:01.654] - [INFO] - [id: 0x2679910f, /127.0.0.1:3279 => localhost/127.0.0.1:19999] CONNECTED: localhost/127.0.0.1:19999
21/09/2014 01:58:02.514] - [INFO] - writing response back for turn 1 for player 1
21/09/2014 01:58:02.526] - [INFO] - message sent: took 79,045,733
21/09/2014 01:59:34.409] - [SEVERE] - Exception thrown in getMove on turn 2
java.lang.NullPointerException
	at GameMap.bfs(GameMap.java:254)
	at GameMap.whichPath(GameMap.java:112)
	at PlayerAI.getMove(PlayerAI.java:56)
	at ClientHandler.messageReceived(ClientHandler.java:85)
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
	at ClientHandler.handleUpstream(ClientHandler.java:40)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:296)
	at org.jboss.netty.handler.codec.oneone.OneToOneDecoder.handleUpstream(OneToOneDecoder.java:70)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:296)
	at org.jboss.netty.handler.codec.frame.FrameDecoder.unfoldAndFireMessageReceived(FrameDecoder.java:462)
	at org.jboss.netty.handler.codec.frame.FrameDecoder.callDecode(FrameDecoder.java:443)
	at org.jboss.netty.handler.codec.frame.FrameDecoder.messageReceived(FrameDecoder.java:310)
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:559)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:268)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:255)
	at org.jboss.netty.channel.socket.oio.OioWorker.process(OioWorker.java:71)
	at org.jboss.netty.channel.socket.oio.AbstractOioWorker.run(AbstractOioWorker.java:73)
	at org.jboss.netty.channel.socket.oio.OioWorker.run(OioWorker.java:51)
	at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108)
	at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

21/09/2014 01:59:34.423] - [INFO] - writing response back for turn 2 for player 1
21/09/2014 01:59:34.429] - [INFO] - message sent: took 91,767,955,003
